### PR TITLE
Support export logging

### PR DESCRIPTION
**Pull request by superlinkx**
## Description

- Add support for conditional auditing
- Add support for including RequestURL in audit logs
- Add better support for pprof
- Fix potential for deadlocking in analysis

## Motivation and Context

Required for some BHE related audit logging. Will likely come in handy as we expand audit logging throughout BHCE as well

## How Has This Been Tested?

- Local testing shows these changes are working as expected
- All automated tests run successfully

## Types of changes

-   [x] Chore (a change that does not modify the application functionality)
-   [x] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [ ] Documentation updates are needed, and have been made accordingly.
-   [ ] I have added and/or updated tests to cover my changes.
-   [x] All new and existing tests passed.
-   [ ] My changes include a database migration.

### CODE REVIEW

File: cmd/api/src/api/middleware/middleware.go [hunk 1]
```diff
@@ -139,7 +140,8 @@ func ContextMiddleware(next http.Handler) http.Handler {
 					Scheme: getScheme(request),
 					Host:   request.Host,
 				},
-				RequestIP: parseUserIP(request),
+				RequestedURL: model.AuditableURL(request.URL.String()),
```
Comment (superlinkx) [inline cmd/api/src/api/middleware/middleware.go:143]: Pull the request URL for auditing purposes

File: cmd/api/src/api/tools/trace.go
Comment (superlinkx) [inline cmd/api/src/api/tools/trace.go:1]: Remove dead code

File: cmd/api/src/daemons/api/toolapi/api.go
Comment (superlinkx) [inline cmd/api/src/daemons/api/toolapi/api.go:1]: Replace our bespoke trace endpoint in the toolapi with the proper `pprof` compatible tools for easier use

File: cmd/api/src/database/audit.go
Comment (superlinkx) [inline cmd/api/src/database/audit.go:1]: Kind of a dumb name, but allows us to add conditional audit logging a bit easier

File: cmd/api/src/model/audit.go
Comment (superlinkx) [inline cmd/api/src/model/audit.go:1]: AuditableURL was needed to support passing the request url to the audit log logic that requires an `Auditable` interface

File: packages/go/analysis/ad/membership.go
Comment (superlinkx) [inline packages/go/analysis/ad/membership.go:1]: Work by @zinic to plug a deadlock condition I hit locally. This cleans up the logic of this function in general, but should be reviewed carefully to ensure nothing was missed. Tests fine locally now

File: packages/go/slicesext/slicesext_test.go
Comment (superlinkx) [inline packages/go/slicesext/slicesext_test.go:1]: Remove dead code

File: cmd/api/src/database/audit.go [hunk 2]
```diff
@@ -103,6 +103,14 @@ func (s *BloodhoundDB) ListAuditLogs(before, after time.Time, offset, limit int,
 	return auditLogs, int(count), CheckError(result)
 }
 
+func (s *BloodhoundDB) MaybeAuditableTransaction(ctx context.Context, auditDisabled bool, auditEntry model.AuditEntry, f func(tx *gorm.DB) error, opts ...*sql.TxOptions) error {
```
[the remark this answers is not in the document]
Comment (superlinkx) [reply cmd/api/src/database/audit.go:106]: It's very much a "I hope we refactor this out, it's just too complicated for this effort". Maybe having such a dumb name will spur that fix down the road lol

**Review by juggernot325 — APPROVED** [review]
LGTM

**Pull request merged**